src/components/SchemaViewer.svelte

```diff
@@ -11,15 +11,32 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames) => {
       let modifiedNode = node;
+      const parentNames =
+        parentNodeNames.length > 0 ? [...parentNodeNames] : [];
 
       if (modifiedNode.fields) {
-        modifiedNode.fields.forEach(addVisibility);
+        if (parentNames.length > 0) {
+          parentNames.push(`${parentNames.slice(-1)}.${modifiedNode.name}`);
+        } else {
+          parentNames.push(`${modifiedNode.name}`);
+        }
+
+        modifiedNode.fields.forEach((field) => {
+          let modifiedNodeField = field;
+          modifiedNodeField.parentNames = parentNames;
+          return addVisibility(modifiedNodeField, parentNames);
+        });
       }
       modifiedNode.visible =
         filterTerms.length === 0 ||
-        filterTerms.every((term) => modifiedNode.name.includes(term));
+        filterTerms.every((term) =>
+          modifiedNode.parentNames && modifiedNode.parentNames.length > 0
+            ? modifiedNode.parentNames.some((val) => val.includes(term)) ||
+              modifiedNode.name.includes(term)
+            : modifiedNode.name.includes(term)
```

Comment:
If I'm reading this correctly, `parentNames` will always be greater than 0 and include the name of the current node, so this logic can be simplified.

Reply:
My previous comment still applies.

```js
           filterTerms.every((term) => parentNames.some(...) || modifiedNode.name.includes(term));
```